packages/dynamic-import/dynamic-versions.js

```diff
@@ -30,3 +30,54 @@ exports.get = function (id) {
 
   return version;
 };
+
+function flattenModuleTree(tree) {
+  var parts = [""];
+  var result = Object.create(null);
+
+  function walk(t) {
+    if (t && typeof t === "object") {
+      Object.keys(t).forEach(function (key) {
+        parts.push(key);
+        walk(t[key]);
+        parts.pop();
+      });
+    } else if (typeof t === "string") {
+      result[parts.join("/")] = t;
+    }
+  }
+
+  walk(tree);
+
+  return result;
+}
+
+// If Package.appcache is loaded, preload additional modules after
+// the core bundle has been loaded.
+function precacheOnLoad(event) {
+  // Check inside onload to make sure Package.appcache has had a
+  // chance to become available.
+  if (!Package.appcache) {
+    return
+  }
+  // prefetch in chunks to reduce overhead
+  // If we call module.prefetch(id) multiple times in the same tick of
+  // the event loop, all those modules will be fetched in one request.
+  function prefetchInChunks(modules, amount) {
+    Promise.all(modules.splice(0, amount).map(function (id) {
+      return module.prefetch(id)
+    })).then(function () {
+      if (modules.length > 0) {
+        prefetchInChunks(modules, amount);
+      }
+    })
+  }
+  // Get a flat array of modules and start prefetching.
+  prefetchInChunks(Object.keys(flattenModuleTree(versions)), 20);
```

Comment:
Should `amount` be configurable?

Reply:
That's a good question, but I'd like to wait and see if people need this. That said, I think I am going to bump this up to 50.